monai/data/dataset.py

```diff
@@ -1061,12 +1061,11 @@ def _transform(self, index: int):
         if not self.transform:
             return data
 
-        transformed_data = apply_transform(self.transform, data)
+        result = apply_transform(self.transform, data)
 
-        if not isinstance(transformed_data, dict):
-            raise AssertionError("With a dict supplied to apply_transform a single dict return is expected.")
-
-        return transformed_data
+        if isinstance(result, dict) or (isinstance(result, list) and isinstance(result[0], dict)):
```

Comment:
This line can be simplified into:
```
if isinstance(result, (dict, list)) and isinstance(result[0], dict)):
```

Reply:
Hi @yiheng-wang-nv ,

Thanks for your review, I think your suggestion may not be correct if the `result` is a dict, result[0] is not dict.

Reply:
Oh, sorry you're right.